src/Kunstmaan/NodeSearchBundle/Services/SearchService.php

```diff
@@ -120,13 +125,28 @@ public function search()
         $request = $this->requestStack->getCurrentRequest();
 
         // Retrieve the current page number from the URL, if not present of lower than 1, set it to 1
+        $pageNumber = $this->getRequestedPage($request);
         $entity = $request->attributes->get('_entity');
 
-        $pageNumber = $this->getRequestedPage($request);
-        $searcher   = $this->container->get($entity->getSearcher());
+        if (!empty($this->searchers)) {
+            foreach ($this->searchers as $possibleSearcher) {
+                if ($entity->getSearcher() === \get_class($possibleSearcher)) {
+                    $searcher = $possibleSearcher;
+                }
```

Comment:
Maybe add a `break;` when we found a searcher as there is no need to keep looping then